bam/reader.go

```diff
@@ -138,11 +136,14 @@ func (b *buffer) readUint32() uint32 {
 	return binary.LittleEndian.Uint32(b.bytes(4))
 }
 
-func newBuffer(r *bgzf.Reader, buf [4]byte) (*buffer, error) {
+func newBuffer(br *Reader, buf [4]byte) (*buffer, error) {
```

Comment:
*A* buf is needed, but you don't need to pass it in. Just use one off the stack.